File: solutions/1748/java/Solution.java
// 1748. Sum of Unique Elements
import java.util.HashMap;

class Solution
{
	public int sumOfUnique(int[] nums)
	{
		HashMap<Integer, Integer> numberOccurences = new HashMap<Integer, Integer>();

		for (int i = 0; i < nums.length; ++i)
		{
			if (numberOccurences.containsKey(nums[i]))
			{
				numberOccurences.put(nums[i], numberOccurences.get(nums[i]) + 1);
			}
			else
			{
				numberOccurences.put(nums[i], 1);
			}
		}

		int sum = 0;

		for (Integer key : numberOccurences.keySet())
		{
			if (numberOccurences.get(key) != 1)
			{
				continue;
			}

			sum += key;
		}

		return sum;
	}
}

File: solutions/1748/js/solution.js
// 1748. Sum of Unique Elements
var sumOfUnique = function(nums)
{
	let numberOccurences = new Map();

	for (let i = 0; i < nums.length; ++i)
	{
		if (numberOccurences.has(nums[i]))
		{
			numberOccurences.set(nums[i], numberOccurences.get(nums[i]) + 1);
		}
		else
		{
			numberOccurences.set(nums[i], 1);
		}
	}

	let sum = 0;

	for (const element of numberOccurences)
	{
		if (element[1] != 1)
		{
			continue;
		}

		sum += element[0];
	}

	return sum;
};